File: internal/content/testdata/bench/rust/nc-lca-bst.rs
use std::rc::Rc;
use std::cell::RefCell;

fn lca_bst(root: Option<Rc<RefCell<TreeNode>>>, p: i64, q: i64) -> i64 {
    let mut node = root;
    while let Some(n) = node {
        let v = n.borrow().val as i64;
        if p < v && q < v {
            let left = n.borrow().left.clone();
            node = left;
        } else if p > v && q > v {
            let right = n.borrow().right.clone();
            node = right;
        } else {
            return v;
        }
    }
    -1
}

File: internal/content/testdata/bench/javascript/nc-lca-bst.js
function lca_bst(root, p, q) {
    let node = root;
    while (node !== null) {
        if (p < node.val && q < node.val) {
            node = node.left;
        } else if (p > node.val && q > node.val) {
            node = node.right;
        } else {
            return node.val;
        }
    }
    return -1;
}
